### PR TITLE
Upgrade to the latest boost version (1.80)

### PR DESCRIPTION
**Pull request by toly-kournik**
* **Problem Overview**  
Upgrade yaml-cpp to a newer version (0.7.0) as it used a non-compatible boost version and now it's a boost free version. Upgrade libthrift to v0.13.0
Upgrade libjaegertracing-cpp to v0.9.0

* **Testing Done**  
unittests, concord CI, maestro 5 days

### CODE REVIEW

**Comment by eyalrund** [comment]
please make sure you also test it with a long run on maestro

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,7 +40,7 @@ option(USE_JSON "Enable use of JSON library" ON)
 option(USE_HTTPLIB "Enable use of httplib" ON)
 option(USE_GRPC "Enable GRPC and Protobuf" ON)
 option(USE_OPENSSL "Enable use of OpenSSL" ON)
-option(BUILD_THIRDPARTY "Wheter to build third party librarie or use preinstalled ones" OFF)
+option(BUILD_THIRDPARTY "Wheter to build third party libraries or use preinstalled ones" OFF)
```
Comment (eyalrund) [inline CMakeLists.txt:43]: typo: whether

File: client/clientservice/CMakeLists.txt [hunk 2]
```diff
@@ -1,13 +1,13 @@
 find_package(GRPC REQUIRED)
 find_package(Boost ${MIN_BOOST_VERSION} COMPONENTS program_options REQUIRED)
 find_package(jaegertracing REQUIRED)
```
Comment (WildFireFlum) [inline client/clientservice/CMakeLists.txt:3]: Shouldn't there be a version here as well?

File: client/thin-replica-client/src/trc_hash.cpp [hunk 3]
```diff
@@ -52,19 +52,19 @@ static string hashUpdateFromEntryHashes(uint64_t block_id, const map<string, str
   string concatenated_entry_hashes;
   concatenated_entry_hashes.reserve(sizeof(block_id) + 2 * entry_hashes.size() * kThinReplicaHashLength);
 
-#ifdef BOOST_LITTLE_ENDIAN
+#ifdef BOOST_ENDIAN_LITTLE_BYTE
   concatenated_entry_hashes.append(reinterpret_cast<const char*>(&block_id), sizeof(block_id));
-#else  // BOOST_LITTLE_ENDIAN not defined in this case
-#ifndef BOOST_BIG_ENDIAN
+#else  // BOOST_ENDIAN_LITTLE_BYTE not defined in this case
+#ifndef BOOST_ENDIAN_BIG_BYTE
```
Comment (WildFireFlum) [inline client/thin-replica-client/src/trc_hash.cpp:58]: What is the meaning of the new names?
Was the rename necessary?

File: examples/client/CMakeLists.txt [hunk 4]
```diff
@@ -1,5 +1,7 @@
 project (test_client LANGUAGES CXX)
 
+find_package(yaml-cpp "0.7.0" REQUIRED)
```
Comment (WildFireFlum) [inline examples/client/CMakeLists.txt:3]: Isn't there a way of reusing the same version string in all of the cmakelists?

File: examples/replica/CMakeLists.txt [hunk 5]
```diff
@@ -1,5 +1,7 @@
 project (test_replica LANGUAGES CXX)
 
+find_package(yaml-cpp "0.7.0" REQUIRED)
```
Comment (WildFireFlum) [inline examples/replica/CMakeLists.txt:3]: Same question regarding the version

**Review by WildFireFlum — APPROVED** [review]

File: client/thin-replica-client/src/trc_hash.cpp (hunk 3 above)
Comment (toly-kournik) [reply client/thin-replica-client/src/trc_hash.cpp:58]: definition move from `<boost/detail/endian.hpp>` to `<boost/predef/other/endian.h>`
I was sure I redefined them in our  `util/endianess.hpp` header, but it disappeared somehow.
Will fix.

File: examples/client/CMakeLists.txt (hunk 4 above)
Comment (toly-kournik) [reply examples/client/CMakeLists.txt:3]: yes, like in boost, will fix

File: client/clientservice/CMakeLists.txt (hunk 2 above)
Comment (toly-kournik) [reply client/clientservice/CMakeLists.txt:3]: Not dealing with it right now, my intention is to remove jaeger from the library

File: CMakeLists.txt (hunk 1 above)
Comment (toly-kournik) [reply CMakeLists.txt]: fixed

**Review by eyalrund — APPROVED** [review]

**Pull request merged**